lib/babe/babe.go

```diff
@@ -154,65 +158,82 @@ func NewService(cfg *ServiceConfig) (*Service, error) {
 	return babeService, nil
 }
 
-func (b *Service) setEpochData(cfg *ServiceConfig, genCfg *types.BabeConfiguration) (err error) {
-	b.epochData = &epochData{
-		randomness: genCfg.Randomness,
+func (b *Service) setupParameters(cfg *ServiceConfig) error {
+	var err error
+	b.epochData = &epochData{}
+
+	epochData, err := b.epochState.GetLatestEpochData()
+	if err != nil {
+		return err
+	}
+
+	b.epochData.randomness = epochData.Randomness
+
+	configData, err := b.epochState.GetLatestConfigData()
+	if err != nil {
+		return err
 	}
 
 	// if slot duration is set via the config file, overwrite the runtime value
-	if cfg.SlotDuration > 0 {
+	if cfg.SlotDuration > 0 && cfg.IsDev { // TODO: remove this, needs to be set via runtime
```

Comment:
Changing this to a switch case may clean up the code.

Reply:
you mean something like:
```
switch cfg.IsDev {
case true:
   if cfg.SlotDuration > 0 { // do stuff }
case false:
   if cfg.SlotDuration > 0 { // error }
}
``` 
?

Reply:
I don't mind nested switches as well. But a switch on `cfg.IsDev` would make it much easier to read.

Reply:
```
	switch {
	case cfg.SlotDuration > 0 && cfg.IsDev:
		b.slotDuration, err = time.ParseDuration(fmt.Sprintf("%dms", cfg.SlotDuration))
	case cfg.SlotDuration > 0 && !cfg.IsDev:
		err = errors.New("slot duration modified in config for non-dev chain")
	default:
		b.slotDuration, err = b.epochState.GetSlotDuration()
	}
```

You can do this for the below code as well.

Reply:
update